File: docs/src/test/java/jdocs/actor/fsm/Events.java
/*
 * Licensed to the Apache Software Foundation (ASF) under one or more
 * license agreements; and to You under the Apache License, version 2.0:
 *
 *   https://www.apache.org/licenses/LICENSE-2.0
 *
 * This file is part of the Apache Pekko project, which was derived from Akka.
 */

/*
 * Copyright (C) 2009-2022 Lightbend Inc. <https://www.lightbend.com>
 */

package jdocs.actor.fsm;

import org.apache.pekko.actor.ActorRef;
import java.util.List;

public class Events {

  public
  // #simple-events
  static final class SetTarget {
    private final ActorRef ref;

    public SetTarget(ActorRef ref) {
      this.ref = ref;
    }

    public ActorRef getRef() {
      return ref;
    }
    // #boilerplate

    @Override
    public String toString() {
      return "SetTarget{" + "ref=" + ref + '}';
    }
    // #boilerplate
  }

  // #simple-events
  public
  // #simple-events
  static final class Queue {
    private final Object obj;

    public Queue(Object obj) {
      this.obj = obj;
    }

    public Object getObj() {
      return obj;
    }
    // #boilerplate

    @Override
    public String toString() {
      return "Queue{" + "obj=" + obj + '}';
    }
    // #boilerplate
  }

  // #simple-events
  public
  // #simple-events
  static final class Batch {
    private final List<Object> list;

    public Batch(List<Object> list) {
      this.list = list;
    }

    public List<Object> getList() {
      return list;
    }
    // #boilerplate

    @Override
    public boolean equals(Object o) {
      if (this == o) return true;
      if (o == null || getClass() != o.getClass()) return false;

      Batch batch = (Batch) o;

      return list.equals(batch.list);
    }

    @Override
    public int hashCode() {
      return list.hashCode();
    }

    @Override
    public String toString() {
      final StringBuilder builder = new StringBuilder();
      builder.append("Batch{list=");
      list.stream()
          .forEachOrdered(
              e -> {
                builder.append(e);
                builder.append(",");
              });
      int len = builder.length();
      builder.replace(len, len, "}");
      return builder.toString();
    }
    // #boilerplate
  }

  // #simple-events
  public
  // #simple-events
  static enum Flush {
    Flush
  }
  // #simple-events
}

File: cluster-sharding/src/main/scala/org/apache/pekko/cluster/sharding/internal/jfr/Events.scala
/*
 * Licensed to the Apache Software Foundation (ASF) under one or more
 * license agreements; and to You under the Apache License, version 2.0:
 *
 *   https://www.apache.org/licenses/LICENSE-2.0
 *
 * This file is part of the Apache Pekko project, which was derived from Akka.
 */

/*
 * Copyright (C) 2020-2022 Lightbend Inc. <https://www.lightbend.com>
 */

package org.apache.pekko.cluster.sharding.internal.jfr

import jdk.jfr.{ Category, Event, Label, StackTrace, Timespan }

import org.apache.pekko.annotation.InternalApi

// requires jdk9+ to compile
// for editing these in IntelliJ, open module settings, change JDK dependency to 11 for only this module

/** INTERNAL API */

@InternalApi
@StackTrace(false)
@Category(Array("Pekko", "Sharding", "Shard")) @Label("Remember Entity Operation")
final class RememberEntityWrite(@Timespan(Timespan.NANOSECONDS) val timeTaken: Long) extends Event

@InternalApi
@StackTrace(false)
@Category(Array("Pekko", "Sharding", "Shard")) @Label("Remember Entity Add")
final class RememberEntityAdd(val entityId: String) extends Event

@InternalApi
@StackTrace(false)
@Category(Array("Pekko", "Sharding", "Shard")) @Label("Remember Entity Remove")
final class RememberEntityRemove(val entityId: String) extends Event

@InternalApi
@StackTrace(false)
@Category(Array("Pekko", "Sharding", "Shard")) @Label("Passivate")
final class Passivate(val entityId: String) extends Event

@InternalApi
@StackTrace(false)
@Category(Array("Pekko", "Sharding", "Shard")) @Label("Passivate Restart")
final class PassivateRestart(val entityId: String) extends Event
